browser/speedreader/speedreader_tab_helper.cc

```diff
@@ -302,6 +344,36 @@ void SpeedreaderTabHelper::DidStopLoading() {
   }
 }
 
+void SpeedreaderTabHelper::DOMContentLoaded(
+    content::RenderFrameHost* render_frame_host) {
+  if (!PageWantsDistill(distill_state_))
+    return;
+
+  constexpr int kIsolatedWorldId = content::ISOLATED_WORLD_ID_CONTENT_END + 1;
+
+  constexpr const char16_t kAddShowOriginalPageLink[] =
+      uR"js(
+    (function() {
+      const link = document.getElementById('show-original-page');
+      if (!link)
+        return;
+      link.text = '$1';
+      link.addEventListener('click', (e) => {
+        window.speedreader.showOriginalPage();
```

Comment:
i assume js on the page isn't able to call `window.speedreader` since it's in an isolated world, but could page js call `document.getElementById('show-original-page').click()` ? just wondering if a page can abuse this feature to force the user to always see the original page.

Reply:
NVM i am told that pages in speedreader mode can't run JS